pandas/_libs/index_class_helper.pxi.in

```diff
@@ -60,7 +60,16 @@ cdef class {{name}}Engine(IndexEngine):
 
         # A view is needed for some subclasses, such as PeriodEngine:
         values = self._get_index_values().view('{{dtype}}')
-        indexer = values == val
+        try:
+            with warnings.catch_warnings():
```

Comment:
Any idea what the perf impact it here?  I don't recall how expensive getting this context manager is.

Reply:
No idea.  A couple options: 1) not-suppress the warning, 2) write a cython function to check for cases when the warning _would_ be emitted and skip the comparison in that case

Reply:
an option here is to do this

```
if isinstance(val, str):
    val = np.array([val], dtype=object)
```

Reply:
Some timings

```python
In [2]: %%timeit
   ...: with warnings.catch_warnings():
   ...:     pass
   ...:
2.19 µs ± 89.6 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [3]: %%timeit
   ...: with warnings.catch_warnings():
   ...:     warnings.filterwarnings("ignore", category=FutureWarning)
   ...:
   ...:
3.96 µs ± 64.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

```

No idea if that's too large here or not.

Reply:
That's not terrible for not-in-a-loop code.  The cython-provided numpy.pxd file has a UFUNC_ERR_IGNORE that makes me think it may be possible to do something similar in cython-space, but so far no helpful documentation on how to make that a reality.

Reply:
> That's not terrible for not-in-a-loop code.

Agreed. Doesn't seem worth worrying about.